From: Hailey Mothershead <hailmo@amazon.com>
Date: Sat, 17 Jun 2023 16:12:15 +0000
Subject: crypto: tcrypt.c - Add selftest for ffdhe algorithims

For fips 140-3 at least one self-test for this algorithim must be run on
boot. Add an invocation here.
---
 crypto/tcrypt.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/crypto/tcrypt.c b/crypto/tcrypt.c
--- a/crypto/tcrypt.c
+++ b/crypto/tcrypt.c
@@ -1702,6 +1702,10 @@ static int do_test(const char *alg, u32 type, u32 mask, int m, u32 num_mb)
 		ret = min(ret, tcrypt_test("gcm(aria)"));
 		break;
 
+	case 59:
+		ret = min(ret, tcrypt_test("ffdhe2048(dh)"));
+		break;
+
 	case 100:
 		ret = min(ret, tcrypt_test("hmac(md5)"));
 		break;

